### DISCUSSION TITLE
Linux games on 16k pages?

### DISCUSSION THREAD

**Discussion by DarkShadow44** [ptitSeb/box64#1522] [General]
The latest release notes say

> Linux Games can now load even on a 16K pagesize machine like a mac M1, and games (including Unity3D ones) can be played with no issues.

I'm curious as to what exactly changed? I already ran factorio (x86_64) using the last version (0.2.6) of box64.
Also, what exactly do you do to allow loading of 4k page binaries on 16k pages?

**Comment by ksco** [comment]
The elf loader, here: https://github.com/ptitSeb/box64/blob/d38f659f428544de524fe6f2a49355a104055555/src/elfs/elfloader.c#L293

**Reply by DarkShadow44** [reply]
So the only change is that it is now dynamic and you don't need to re-compile for 16k pages?

**Comment by ptitSeb** [comment]
> So the only change is that it is now dynamic and you don't need to re-compile for 16k pages?

Not only that. All the elfloader code as been refactored to better handle the non-4k pagesize cases. It was not working all the time in previous version. Now it just work.
Some games that don't use much emulated libs were working, but most were just failling at load.

**Reply by DarkShadow44** [reply]
Alright, thanks. I was just confused since it seemed to work for me in the last version as well, but glad to hear you managed to get even better compatibility! I'll look at how you did it in hopes I learn something that helps do the same for Wine.
